metaseq/utils.py

```diff
@@ -607,6 +615,29 @@ def get_random_port():
     return port
 
 
+def floating_point_precision_convertor(
+    x, fp16: bool, memory_efficient_fp16: bool, bf16: bool
+):
+    """
+    Convert a tensor x into the desired dtype.
+
+    Also sanity checks combinations of options.
+    """
+    if memory_efficient_fp16:
+        assert not bf16, "Do not combined bf16 with memory_efficient_fp16."
+    if bf16:
+        assert fp16, "Setting --bf16 requires also setting --fp16 for legacy reasons."
```

Comment:
;_;

Reply:
hey, at least it doesn't just silently do the wrong thing :P